lti_consumer/models.py

```diff
@@ -296,3 +315,17 @@ def __str__(self):
 
     class Meta:
         unique_together = (('line_item', 'user_id'),)
+
+
+@receiver(post_save, sender=LtiAgsScore)
+def update_student_grade(sender, instance, **kwargs):
+    if instance.grading_progress == LtiAgsScore.FULLY_GRADED:
+
+        # get lti config
+        lti_config = instance.line_item.lti_configuration
+
+        # find user
+        user = get_user_from_external_user_id(instance.user_id)
+
+        # save grade to xblock, this rebounds user with xblock internally
+        lti_config.block.set_user_module_score(user, instance.score_given, instance.score_maximum, instance.comment)
```

Comment:
From the internal ticket:

>Need a bit of help. Here, I am trying to save the score to xblock (Hopefully I am using the right method). But it is giving the following error -
``` 
Traceback (most recent call last):
--
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
response = get_response(request)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
response = self.process_exception_by_middleware(e, request)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/usr/lib/python3.8/contextlib.py", line 75, in inner
return func(*args, **kwds)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/admin/options.py", line 606, in wrapper
return self.admin_site.admin_view(view)(*args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/utils/decorators.py", line 142, in _wrapped_view
response = view_func(request, *args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
response = view_func(request, *args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/admin/sites.py", line 223, in inner
return view(request, *args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1648, in change_view
return self.changeform_view(request, object_id, form_url, extra_context)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/utils/decorators.py", line 45, in _wrapper
return bound_method(*args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/utils/decorators.py", line 142, in _wrapped_view
response = view_func(request, *args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1529, in changeform_view
return self._changeform_view(request, object_id, form_url, extra_context)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1572, in _changeform_view
self.save_model(request, new_object, form, not add)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1088, in save_model
obj.save()
File "/edx/src/xblock-lti-consumer/lti_consumer/models.py", line 306, in save
super().save(*args, **kwargs)
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/db/models/base.py", line 743, in save
self.save_base(using=using, force_insert=force_insert,
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/db/models/base.py", line 791, in save_base
post_save.send(
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 173, in send
return [
File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/dispatch/dispatcher.py", line 174, in <listcomp>
(receiver, receiver(signal=self, sender=sender, **named))
File "/edx/src/xblock-lti-consumer/lti_consumer/models.py", line 331, in update_student_grade
lti_config.block.set_user_module_score(user, instance.score_given, instance.score_maximum, instance.comment)
File "/edx/src/xblock-lti-consumer/lti_consumer/lti_xblock.py", line 1375, in set_user_module_score
self.runtime.rebind_noauth_module_to_user(self, user)
File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 2025, in __getattr__
return getattr(self._descriptor_system, name)
AttributeError: 'CachingDescriptorSystem' object has no attribute 'rebind_noauth_module_to_user'
```

I'll look at this today and try to test a few things.
This has probably something to do with how the block is loaded by the model.

Reply:
This PR: https://github.com/edx/xblock-lti-consumer/pull/116 fixes the issue.

Reply:
Not using that approach, instead using ``lms.djangoapps.grades.api.signals.grades_signals.SCORE_PUBLISHED``